cirq-core/cirq/sim/act_on_args.py

```diff
@@ -113,14 +115,34 @@ def _perform_measurement(self, qubits: Sequence['cirq.Qid']) -> List[int]:
         """Child classes that perform measurements should implement this with
         the implementation."""
 
-    def copy(self: TSelf) -> TSelf:
-        """Creates a copy of the object."""
+    def copy(self: TSelf, reuse_buffer: bool = False) -> TSelf:
```

Comment:
We managed to make this non breaking with some reflection. Hence the test that still works. Warning messages are appropriately emitted when the new argument does not exist (and there are tests around that as well).

This could be a good reference for doing such things.

Reply:
Yes, if we check the signature before actually calling the `copy` (like the case in `SimulatorBase._run`), we can avoid errors.